arch/Kconfig

```diff
@@ -515,8 +518,8 @@ config MPU_GAP_FILLING
 
 menuconfig FLOAT
 	bool "Floating point"
-	depends on CPU_HAS_FPU
 	depends on ARM || X86 || ARC
+	depends on CPU_HAS_FPU || ARCH_HAS_SOFTWARE_FP_ABI
 	help
 	  This option allows threads to use the floating point registers.
```

Comment:
Guess this text should be updated to reflect that it can also be enabled if architecture supports software based floating point operations.

In that case there is not registers.